exporter/elasticsearchexporter/exporter_test.go

```diff
@@ -216,7 +217,10 @@ func TestExporterLogs(t *testing.T) {
 		server := newESTestServer(t, func(docs []itemRequest) ([]itemResponse, error) {
 			rec.Record(docs)
 
-			assert.Equal(t, "logs-record.dataset-resource.namespace", actionJSONToIndex(t, docs[0].Action))
+			ds := "record.dataset" + strings.Repeat("_", len(disallowedDatasetRunes))
+			ns := "resource.namespace" + strings.Repeat("_", len(disallowedNamespaceRunes))
+
+			assert.Equal(t, fmt.Sprintf("logs-%s-%s", ds, ns), actionJSONToIndex(t, docs[0].Action))
```

Comment:
As exporter_test is like a e2e test for the exporter, I don't expect it to have too much knowledge of private variables of data stream router. There's also an added benefit that it will catch a change if someone fat fingered to delete `-` in `disallowedNamespaceRunes`.
```suggestion
			assert.Equal(t, "logs-record.dataset._______________-resource.namespace.-______________", actionJSONToIndex(t, docs[0].Action))
```